site/content/how-to/maintenance/upgrade-apps-without-downtime.md

```diff
@@ -0,0 +1,127 @@
+
+For example, an application can be exposed using a routing rule like below:
+
+```yaml
+- matches:
+  - path:
+      type: PathPrefix
+      value: /
+  backendRefs:
+  - name: my-app
+    port: 80
+```
+
+{{< note >}}See the [Cafe example](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/examples/cafe-example) for a basic example.{{< /note >}}
+
+The upgrade methods in the next sections cover:
+
+- Rolling deployment upgrades
+- Blue-green deployments
+- Canary releases
+
+## Rolling deployment upgrade
+
+To start a [rolling deployment upgrade](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment), you update the deployment to use the new version tag of the application. As a result, Kubernetes terminates the pods with the old version and create new ones. By default, Kubernetes also ensures that some number of pods always stay available during the upgrade.
+
+This upgrade will add new upstream servers to NGINX and remove the old ones. As long as the number of pods (ready endpoints) during an upgrade does not reach zero, NGINX will be able to proxy traffic, and therefore prevent any downtime.
+
+This method does not require you to update the **HTTPRoute**.
+
+## Blue-green deployments
+
+With this method, you deploy a new version of the application (blue version) as a separate deployment, while the old version (green) keeps running and handling client traffic. Next, you switch the traffic from the green version to the blue. If the blue works as expected, you terminate the green. Otherwise, you switch the traffic back to the green.
+
+There are two ways to switch the traffic:
+
+- Update the service selector to select the pods of the blue version instead of the green. As a result, NGINX Gateway Fabric removes the green upstream servers from NGINX and add the blue ones. With this approach, it is not necessary to update the **HTTPRoute**.
+- Create a separate service for the blue version and update the backend reference in the **HTTPRoute** to reference this service, which leads to the same result as with the previous option.
+
+## Canary releases
+
+To support canary releases, you can implement an approach with two deployments behind the same service (see [Canary deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#canary-deployment) in the Kubernetes documentation). However, this approach lacks precision for defining the traffic split between the old and the canary version. You can greatly influence it by controlling the number of pods (for example, four pods of the old version and one pod of the canary). However, note that NGINX Gateway Fabric uses [`random two least_conn`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#random) load balancing method, which doesn't guarantee an exact split based on the number of pods (80/20 in the given example).
```

Comment:
As a 1 or 2 sentence introduction, it might be useful to define what a canary release is. Same for the previous sections (blue/green, rolling). 

These are terms that haven't been defined.

Reply:
This is the definition of what a rolling update is:
![image](https://github.com/nginxinc/nginx-gateway-fabric/assets/78599298/0ff1a8ae-20d2-4281-a550-d2cc40729227)

Reply:
This is the definition of what a blue-geen deployment is:
![image](https://github.com/nginxinc/nginx-gateway-fabric/assets/78599298/398a5ecf-304c-467c-ac6b-196b29892f4a)

Reply:
added a paragraph explaining canary releases